Sources/DiagramResult.swift

```diff
@@ -589,3 +565,69 @@ extension DiagramResult {
     }
 }
 
+extension DiagramResult: CurrencyScreenDelegate {
+    func transferCurShortName(currency: String) {
+        let copyConverterScreen = ConverterScreen()
+
+        switch self.selectorDiagram {
+            case 1: do {
+                if currency == "" {
```

Comment:
Please wrap the code inside `case something: do {` into a function a call it in 3 places.